include/flox/core/exceptions.hh

```diff
@@ -99,6 +100,9 @@ public:
   //   methods.
   [[nodiscard]] std::string
   what_string() const noexcept;
+
+  [[nodiscard]] nlohmann::json
```

Comment:
Conventionally `to_json` routines are declared outside of a class which is what allows _magic_ `jsonVal.get_to( val )` and `jsonVal.template get<T>()` routines to work.
If a class contains private member variables you may need to add a `friend void to_json( nlohmann::json &, const FloxException & );` declaration to the class.

If you can tweak this declaration you'll be good to go.

```suggestion
  friend void to_json( nlohmann::json & jto, const FloxException & err );
```